include/odp/api/spec/packet.h

```diff
@@ -1140,6 +1179,82 @@ int odp_packet_move_data(odp_packet_t pkt, uint32_t dst_offset,
  */
 
 /**
+ * Packet parse parameters
+ */
+typedef struct odp_packet_parse_param_t {
+	/** Protocol header at parse starting point. Valid values for this
+	 *  field are: ODP_PROTO_ETH, ODP_PROTO_IPV4, ODP_PROTO_IPV6. */
+	odp_proto_t proto;
+
+	/** Continue parsing until this layer. Must be the same or higher
+	 *  layer than the layer of 'proto'. */
+	odp_proto_layer_t layer;
```

Comment:
it looks like 2 enumbs are not needed here. odp_proto_layet_t  layer_start; and  odp_proto_layet_t  layer_end;  and one enum for layers.

Reply:
Layer is not enough information for the parser. It needs to know the first protocol (e.g. MPLS vs IP) to be able to start the parsing.